internal/providersdkv2/provider.go

```diff
@@ -111,24 +131,33 @@ func configure(p *schema.Provider) func(context.Context, *schema.ResourceData) (
 			diags = isHCPOperational()
 		}
 
-		// Sets up HCP SDK client.
-		userAgent := p.UserAgent("terraform-provider-hcp", version.ProviderVersion)
+		clientConfig := clients.ClientConfig{
+			ClientID:       d.Get("client_id").(string),
+			ClientSecret:   d.Get("client_secret").(string),
+			CredentialFile: d.Get("credential_file").(string),
+			SourceChannel:  p.UserAgent("terraform-provider-hcp", version.ProviderVersion),
+		}
 
-		clientID := d.Get("client_id").(string)
-		if clientID == "" {
-			clientID = os.Getenv("HCP_CLIENT_ID")
+		// Check the environment for an HCP Service Principal
+		if clientConfig.ClientID == "" {
+			clientConfig.ClientID = os.Getenv("HCP_CLIENT_ID")
```

Comment:
🤔 I know this wasn't introduced with this PR, but this shouldn't be necessary. The SDK is already [configured with FromEnv()](https://github.com/hashicorp/terraform-provider-hcp/pull/691/files#diff-460f9527216e2f272b53bb012adc914caa31669c76c04e4e76b1f8c4c8ed0f2cR98).